lib/Contracts/IUserPreferences.php

```diff
@@ -29,15 +29,18 @@
 interface IUserPreferences {
 
 	/**
+	 * @param string $userId
 	 * @param string $key
 	 * @param mixed $value
 	 * @return mixed new value
+	 * @throws \OCP\PreConditionNotMetException
```

Comment:
We don't pass $preCondition to setUserValue so the method will not throw a PreConditionNotMetException. I would prefer to not list in phpdocs.